docs/config.md

```diff
@@ -782,6 +808,74 @@ The base64 encoded protobuf describing (and containing) the node's private key.
 
 Type: `string` (base64 encoded)
 
+## `Internal`
+
+### `Internal.Bitswap`
+
+InternalBitswap contains knobs for tuning bitswap resource utilization.
+The knobs (below) document how their value should related to each other.
+Whether their values should be raised or lowered should be determined
+based on the metrics `active_tasks`, `pending_tasks`, `pending_block_tasks` and `active_block_tasks`
+reported by bitswap.
+
+The value of `active_tasks` is capped by `EngineTaskWorkerCount`.
+
+The value of `pending_tasks` is generally capped by the knobs below,
+however its exact maximum value is hard to predict as it depends on task sizes
+as well as number of requesting peers. However, as a rule of thumb,
+during healthy operation this value should oscillate around a "typical" low value
+(without hitting a plateau continuously).
+
+If `pending_tasks` grows and eventually reaches a plateau,
+while at the same time `active_tasks` is at its maximum,
+the node has reached its resource limits and requests are being dropped (or not serviced).
+Raising resource limits (using the knobs below) could help, assuming the hardware can support the new limits.
+
+The value of `active_block_tasks` is capped by `EngineBlockstoreWorkerCount`.
+
+The value of `pending_block_tasks` is indirectly capped by `active_tasks`, but cannot be predicted
+as it depends on the number of blocks involved in a peer task which can vary.
+
+If the value of `pending_block_tasks` is observed to grow,
+while `active_block_tasks` is at its maximum, there is indication that the number of
+available block tasks is creating a bottleneck (either due to high-latency block operations,
+or due to high number of block operations per bitswap peer task).
+In such cases, try increasing the `EngineBlockstoreWorkerCount`.
+
+#### `Internal.Bitswap.TaskWorkerCount`
+
+Number of threads sending outgoing messages.
+Used to thottle the number of concurrent send operations.
+
+Type: `int`
+
+#### `Internal.Bitswap.EngineBlockstoreWorkerCount`
+
+Number of threads for blockstore operations.
+Used to throttle the number of concurrent requests to the block store.
+This number should generally be a low multiple (e.g. 4) of the `TaskWorkerCount`,
```

Comment:
> This number should generally be a low multiple (e.g. 4) of the `TaskWorkerCount`,

Where'd you get this from? I think the defaults we run with are 8 task workers and 128 blockstore workers

Reply:
I simplified this. I used the "multiple" explanation initially because the TaskWorkerCount threads generate block tasks, hence a multiple. And also the PR you handed off to me had it parameterized like that. But for now I am simply saying it depends on your hardware. Also valid.